src/coord/timestamp.rs

```diff
@@ -155,6 +156,14 @@ lazy_static! {
         &["topic", "source_id", "partition_id"]
     )
     .unwrap();
+
+      /// The max timestamp that we have allocated for this partition
+    static ref MAX_TIMESTAMP: IntGaugeVec = register_int_gauge_vec!(
+        "mz_max_timestamp",
+        "The high watermark for a partition, the maximum offset that we could hope to ingest",
+        &["source_id", "partition_id"]
```

Comment:
the fact that this is for a partition suggests that it only works for kafka, and should be `mz_kafka_max_timestamp` and is possibly actually the same thing as `mz_kafka_partition_offset_max`? How is it different?

Is it just for situations where we don't know the topic ID? If that's the case could we just use `"<unknown>"` or something as the topic name?

Reply:
Partition ID applies to all sources (it would just all map to the same source for files and ocf sources). For Kinesis and Kafka, it would be different.

We do know the topic id, but it's not passed in as an argument to the function, so I just left it at source id (source id is what is unique anyways, it feels better to use that as the main naming source?)

mz_kafka_partition_offset_max is the max offset (ak the max number of records). mz_max_timestamp is the maximum timestamp we have allocated. Basically I want to be able to track the difference between what timestamp is "available", and what timestamp wee are done "processing"